### DISCUSSION TITLE
Response status message pending

### DISCUSSION THREAD

**Discussion by caturhidayat** [kirimwa-id/forum#152] [Tanya Jawab] [answered]
Dear Mas @rioastamal 

Sebelumnya terimakasih untuk mas rio yang sudah menyediakan kirimwa.id ini secara gratis.
Disini saya masih belajar untuk penggunaan API dan webhook. 
Ide dari project saya adalah membuat kirim WA otomatais ketika ada daftar kontak yang ulangtahun.
Saya menggunakan cron untuk mengecek database apakah ada yang ulang tahun atau tidak.
Untuk fungsi kirim WA nya berhasil mas, akan tetapi saat saya mau ambil response status dari webhook nya kok tetap pending ya? apakah ada yang salah dengan script saya?
Berikut kode saya : 
<img width="728" alt="CleanShot 2022-11-14 at 12 09 07@2x" src="https://user-images.githubusercontent.com/94335744/201580841-ef2b6773-82a4-4b46-9ba0-4d7bfdc5c694.png">

Dan ini respon yang saya dapat : 

![CleanShot 2022-11-14 at 12 08 51@2x](https://user-images.githubusercontent.com/94335744/201580923-e184ad0e-52e0-4724-b054-4b838dee2d70.png)

Mohon arahannya mas 🙏🏼

**Comment by rioastamal** [comment, marked as answer]
@caturhidayat, Sudah dijelaskan di [dokumentasi API](https://developer.kirimwa.id/api/v1/#messages) bahwa proses pengiriman bersifat asynchronuous. Jadi setelah request ke API `POST /v1/messages` pesan akan masuk ke status `pending` dahulu dan akan masuk queue/antrian untuk diproses.

Setelah diproses dan pesan dikirim maka status akan diubah ke `success`. Jika anda telah memasukkan webhook maka API KirimWA.id akan mengirim webhook ke URL anda untuk memberi tahu perubahan status dari pesan.

Status dari pesan `kwid-efd3b1e7bcff4890b7b03374d00` adalah `success` jika anda melakukan query ke `GET /v1/messages/kwid-efd3b1e7bcff4890b7b03374d00`.

**Reply by caturhidayat** [reply]
Dear mas @rioastamal ,
Terima kasih untuk jawaban dan penjelasannya, 🙏🏼
